prometheus/gc_collector.go

```diff
@@ -0,0 +1,43 @@
+package prometheus
+
+import (
+	"runtime/debug"
+)
+
+type gcCollector struct {
+	numGC      Gauge
+	pauseTotal Gauge
+}
+
+// NewGCCollector returns a collector which exports metrics about the
+// garbage collection stats of the current process.
+func NewGCCollector() *gcCollector {
+	return &gcCollector{
+		numGC: NewGauge(GaugeOpts{
+			Name: "process_gc_num_total",
+			Help: "Total number of garbage collector incovations for the current process.",
```

Comment:
(you'd have to generate synthetic summary observations otherwise based on the increases since the last scrape, so I think the two counters are good as is)